tests/unit/python/foglamp/services/core/test_server.py

```diff
@@ -0,0 +1,435 @@
+                    assert {'message': 'Service registered successfully', 'id': '1'} == json_response
+                args, kwargs = audit_info_patch.call_args
+                assert 'SRVRG' == args[0]
+                assert {'name': request_data['name']} == args[1]
+        args, kwargs = patch_register.call_args
+        assert (request_data['name'], request_data['type'], request_data['address'], request_data['service_port'], request_data['management_port'], 'http') == args
+
+    async def test_service_not_found_when_unregister(self, client):
+        with patch.object(ServiceRegistry, 'get', side_effect=service_registry_exceptions.DoesNotExist) as patch_unregister:
+            resp = await client.delete('/foglamp/service/blah')
+            assert 404 == resp.status
+            assert 'Service with blah does not exist' == resp.reason
+        args, kwargs = patch_unregister.call_args
+        assert {'idx': 'blah'} == kwargs
+
+    async def test_unregister_service(self, client):
+        async def async_mock():
+            return ""
+
+        service_id = "c6bbf3c8-f43c-4b0f-ac48-f597f510da0b"
+        sname = "name"
+        stype = "Southbound"
+        sprotocol = "http"
+        saddress = "localhost"
+        sport = 1234
+        smgtport = 4321
+        data = []
+        record = ServiceRecord(service_id, sname, stype, sprotocol, saddress, sport, smgtport)
+        data.append(record)
+        Server._storage_client = MagicMock(StorageClient)
+        with patch.object(ServiceRegistry, 'get', return_value=data) as patch_get_unregister:
+            with patch.object(ServiceRegistry, 'unregister') as patch_unregister:
+                with patch.object(AuditLogger, '__init__', return_value=None):
+                    with patch.object(AuditLogger, 'information', return_value=async_mock()) as audit_info_patch:
+                        resp = await client.delete('/foglamp/service/{}'.format(service_id))
+                        assert 200 == resp.status
+                        r = await resp.text()
+                        json_response = json.loads(r)
+                        assert {'id': service_id, 'message': 'Service unregistered'} == json_response
+                    args, kwargs = audit_info_patch.call_args
+                    assert 'SRVUN' == args[0]
+                    assert {'name': sname} == args[1]
+            args1, kwargs1 = patch_unregister.call_args
+            assert (service_id,) == args1
+        args2, kwargs2 = patch_get_unregister.call_args
+        assert {'idx': service_id} == kwargs2
+
+    async def test_ping(self, client):
+        resp = await client.get('/foglamp/service/ping')
+        assert 200 == resp.status
+        r = await resp.text()
+        json_response = json.loads(r)
+        assert 'uptime' in json_response
+        assert 0.0 < json_response["uptime"]
+
+    # FIXME: Not sure if we need this
+    async def test_change(self, client):
```

Comment:
I don;t think its required. as Server is not implemented with FoglampMicroservice abstract class; but we can keep it for now as it, and the test the same way.